### PR TITLE
Use gql renewSession for trashcan page and fix trashcan task parsing

### PR DESCRIPTION
**Pull request by saberlynx**
**Checklist**:

- [ ] Tests
- [ ] [CHANGELOG](https://github.com/greenbone/gsa/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: gsa/src/web/pages/tasks/row.js [hunk 1]
```diff
@@ -110,11 +110,11 @@ const Row = ({
     if (hasValue(observers.users)) {
       obs.user = _('Users {{user}}', {user: observers.users.join(', ')});
     }
-    if (observers.roles.length > 0) {
+    if (observers?.roles?.length > 0) {
```
Comment (bjoernricks) [inline gsa/src/web/pages/tasks/row.js:113]: Would be better to use
```suggestion
    if (isDefined(observers?.roles?.length) && observers.roles.length > 0) {
```
to not rely on js' behavior of comparing undefined with 0.

File: gsa/src/web/pages/tasks/row.js [hunk 2]
```diff
@@ -110,11 +110,11 @@ const Row = ({
...
       const role = observers.roles.map(r => r.name);
       obs.role = _('Roles {{role}}', {role: role.join(', ')});
     }
-    if (observers.groups.length > 0) {
+    if (observers?.groups?.length > 0) {
```
Comment (bjoernricks) [inline gsa/src/web/pages/tasks/row.js:117]:
```suggestion
    if (isDefined(observers?.roles?.length) && observers.roles.length > 0) {
```

File: gsa/src/gmp/model.js [hunk 3]
```diff
@@ -105,8 +104,15 @@ class Model {
       delete copy.in_use;
     }
 
-    if (isDefined(element.owner) && isEmpty(element.owner.name)) {
-      delete copy.owner;
+    if (isDefined(element.owner)) {
+      if (this.entityType === 'task') {
```
Comment (bjoernricks) [inline gsa/src/gmp/model.js:108]: I really don't want such if checking in the generic model class. Why is this needed? Maybe just parse the owner in the task model class before calling super or just re-parse it from the original element.

File: gsa/src/web/pages/extras/trashcanpage.js [hunk 4]
```diff
@@ -15,9 +15,8 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
-import React from 'react';
-
-import {connect} from 'react-redux';
+/* eslint-disable no-shadow */
```
Comment (bjoernricks) [inline gsa/src/web/pages/extras/trashcanpage.js:18]: Please don't disable no-shadow for the whole file. Shadowing a variable may not be obvious sometimes and can lead to strange bugs.

File: gsa/src/web/pages/extras/trashcanpage.js [hunk 5]
```diff
@@ -119,105 +115,85 @@ const separateByUsageType = inputList => {
...
-  componentDidMount() {
-    this.getTrash();
-  }
+  const [trash, setTrash] = useState({});
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState();
 
-  getTrash() {
-    const {gmp} = this.props;
+  const getTrash = useCallback(() => {
```
Comment (bjoernricks) [inline gsa/src/web/pages/extras/trashcanpage.js:126]: Should be renamed to something more meaningful like loadTrashCanData or something like that.

File: gsa/src/web/pages/extras/trashcanpage.js [hunk 6]
```diff
@@ -119,105 +115,85 @@ const separateByUsageType = inputList => {
...
-      .then(this.getTrash)
-      .catch(error => {
-        this.setState({
-          error: error,
+      return gmp.trashcan
+        .restore(entity)
+        .then(getTrash)
+        .catch(error => {
+          setError(error);
         });
```
Comment (bjoernricks) [inline gsa/src/web/pages/extras/trashcanpage.js:156]: Can be simplified if you want.
```suggestion
        .catch(setError);
```

File: gsa/src/web/pages/extras/trashcanpage.js [hunk 7]
```diff
@@ -119,105 +115,85 @@ const separateByUsageType = inputList => {
...
+  const handleDelete = useCallback(
+    entity => {
+      handleInteraction();
+
+      return gmp.trashcan
+        .delete(entity)
+        .then(getTrash)
+        .catch(error => {
+          setError(error);
         });
```
Comment (bjoernricks) [inline gsa/src/web/pages/extras/trashcanpage.js:170]:
```suggestion
        .catch(setError);
```

File: gsa/src/web/pages/extras/trashcanpage.js [hunk 8]
```diff
@@ -119,105 +115,85 @@ const separateByUsageType = inputList => {
...
+  );
 
-  handleEmpty() {
-    const {gmp} = this.props;
-    this.handleInteraction();
+  const handleEmpty = useCallback(() => {
+    handleInteraction();
 
-    this.setState({loading: true});
+    setLoading(true);
```
Comment (bjoernricks) [inline gsa/src/web/pages/extras/trashcanpage.js:178]: I am confused why loading is set during emptying the trash but not when loading the trash data.

File: gsa/src/web/pages/extras/trashcanpage.js [hunk 9]
```diff
@@ -226,344 +202,312 @@ class Trashcan extends React.Component {
...
+        </TableBody>
+      );
+    },
+    [createContentRow],
+  );
+
+  if (!isDefined(trash) && !isDefined(error)) {
+    return <Loading />;
+  } else if (!isDefined(trash) && isDefined(error)) {
+    setTrash({});
```
Comment (bjoernricks) [inline gsa/src/web/pages/extras/trashcanpage.js:313]: I don't think this works. You are changing the state during a react render phase.

File: gsa/src/web/pages/extras/trashcanpage.js (hunk 8 above)
Comment (saberlynx) [reply gsa/src/web/pages/extras/trashcanpage.js:174]: Carelessness. I'll add that in.

File: gsa/src/web/pages/extras/trashcanpage.js (hunk 9 above)
Comment (saberlynx) [reply gsa/src/web/pages/extras/trashcanpage.js]: I have set it to watch for `loading` instead. trash should always be defined since it's initialized as `{}` and never set to undefined in code.

File: gsa/src/web/pages/extras/trashcanpage.js [hunk 10]
```diff
@@ -119,105 +114,81 @@ const separateByUsageType = inputList => {
...
-  }
-
-  handleEmpty() {
-    const {gmp} = this.props;
-    this.handleInteraction();
+  const handleEmpty = useCallback(() => {
+    handleInteraction();
 
-    this.setState({loading: true});
+    setLoading(true);
```
Comment (bjoernricks) [inline gsa/src/web/pages/extras/trashcanpage.js:173]: I still don't get this loading case. Why is loading set to true until the loading is actually started? Maybe just remove setting loading here in handleEmpty
